os/drivers/lcd/mipi_lcd.c

```diff
@@ -38,6 +38,62 @@
 #define CONFIG_LCD_MAXPOWER 100
 #endif
 
+#if defined(CONFIG_LCD_SW_ROTATION)
+#define NUM_OF_LCD_BUFFER	2
+uint8_t *lcd_buffer[NUM_OF_LCD_BUFFER] = { NULL, NULL };	//Two lcd buffers to avoid screen tearing
+int lcd_buffer_index = 0;
+
+static void lcd_rotate_buffer(short int* src, short int* dst)
+{
+	int row;
+	int col;
+	int dst_inc = 2 * CONFIG_LCD_XRES;
+	short int val0;
+	short int val1;
+	short int val2;
+	short int val3;
+	short int *psrc;
+	short int *pdst;
+
+#if defined(CONFIG_LCD_LANDSCAPE)
+	lldbg("Line number: %d\n", __LINE__);
+	for (row = 0; row < CONFIG_LCD_XRES; row += 2) {
+		psrc = src + row * CONFIG_LCD_YRES;
+		pdst = dst + CONFIG_LCD_XRES - row - 2;
+		for (col = 0; col < CONFIG_LCD_YRES; col += 2) {
+			val0 = *(psrc + 0);
+			val1 = *(psrc + 1);
+			val2 = *(psrc + CONFIG_LCD_YRES + 0);
+			val3 = *(psrc + CONFIG_LCD_YRES + 1);
+			psrc += 2;
+			*(pdst + 0) = val2;
+			*(pdst + 1) = val0;
+			*(pdst + CONFIG_LCD_XRES) = val3;
+			*(pdst + CONFIG_LCD_XRES + 1) = val1;
+			pdst += dst_inc;
+		}
+	}
+#elif defined(CONFIG_LCD_RLANDSCAPE)
+	for (row = 0; row < CONFIG_LCD_XRES; row += 2) {
+		psrc = src + row * CONFIG_LCD_YRES;
+		pdst = dst + row + (CONFIG_LCD_YRES - 1) * CONFIG_LCD_XRES;
+		for (col = 0; col < CONFIG_LCD_YRES; col += 2) {
+			val0 = *(psrc + 0);
+			val1 = *(psrc + 1);
+			val2 = *(psrc + CONFIG_LCD_YRES + 0);
+			val3 = *(psrc + CONFIG_LCD_YRES + 1);
+			psrc += 2;
+			*(pdst + 0) = val0;
+			*(pdst + 1) = val2;
+			*(pdst - CONFIG_LCD_XRES) = val1;
+			*(pdst - CONFIG_LCD_XRES + 1) = val3;
+			pdst -= dst_inc;
+		}
+	}
+#endif
```

Comment:
Let's add `#else` with #error NOT Support blah blah orientation message.

Reply:
How about using `#error` instead of debug message.
Why I suggested is
we can't support what user wants in this case. But debug message makes user know it after build, download and running. It's too late (User will feel bad). I think we can notice it to user at build time using `#error`